user_workspaces_server/views.py

```diff
@@ -160,6 +167,10 @@ def put(self, request, workspace_id, put_type=None):
             return JsonResponse({'message': 'Update successful.', 'success': True})
 
         if put_type.lower() == 'start':
+            if not main_storage.is_valid_workspace_path(workspace.file_path):
+                raise APIException('Please contact a system administrator there is a failure with '
```

Comment:
Can't rely on API users to notify you if something goes wrong. For one, the UI many not even show this message from the API. If you want to be notified, you need logging and monitoring on your side, and you should have some tests of the machinery.